### PR TITLE
238 add ability to merge fastqs

### PR DESCRIPTION
**Pull request by SpikyClip**
## PR checklist

- [x] This comment contains a description of changes (with reason).
    - Added the merging of fastq files with the same sample names after the `INPUT_CHECK` process.
        - Installed `cat/fastq` module using `nf-core` tools.
        - Uncommented `validate_unique_samples()` in `check_samplesheet.py` to mark duplicate sample names.
        - Added merging workflow steps derived from [`nf-core/rnaseq`](https://github.com/nf-core/rnaseq/blob/master/workflows/rnaseq.nf#L195-L222).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - The `test` profiles should be sufficient once we add the proper test dataset with duplicate sample names.
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/methylseq/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/methylseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
    - I don't have access to the appropriate test data, would someone be able to help with this?
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
    - NOTE: I get process failures with `NFCORE_METHYLSEQ:METHYLSEQ:PRESEQ_LCEXTRAP` before and after these edits, I presume they are normal but I am not very familiar with this pipeline's `test` results. The pipeline still completes successfully.
- [x] Usage Documentation in `docs/usage.md` is updated.
    - `usage.md` already implied [merging was done](https://github.com/nf-core/methylseq/blob/master/docs/usage.md#multiple-runs-of-the-same-sample).
- [ ] `CHANGELOG.md` is updated.
    - Not sure if I should update the changelog until we are ready to add it to the master branch (and test data has been settled)
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: bin/check_samplesheet.py [hunk 1]
```diff
@@ -211,7 +211,7 @@ def check_samplesheet(file_in, file_out):
                 logger.critical(f"{str(error)} On line {i + 2}.")
                 sys.exit(1)
         # TODO: ADD THIS BACK WHEN WE HAVE LANE MERGING?
```
Comment (ewels) [inline bin/check_samplesheet.py:213]:
```suggestion
        # Ties into FastQ merging process
```

File: workflows/methylseq.nf [hunk 2]
```diff
@@ -82,27 +83,52 @@ workflow METHYLSEQ {
...
+    .map {
+        meta, fastq ->
+            def meta_clone = meta.clone()
+            meta_clone.id = meta_clone.id.split('_')[0..-2].join('_')
+            [ meta_clone, fastq ]
+    }
+    .groupTuple(by: [0])
+    .branch {
+        meta, fastq ->
+            single  : fastq.size() == 1
```
Comment (ewels) [inline workflows/methylseq.nf:96]:
```suggestion
            single: fastq.size() == 1
```

**Review by ewels — APPROVED** [review]
Reading the code, looks good to me! Might just test locally / add a duplicate process to the `-profile test` tests before merging..

**Comment by ewels** [comment]
Just updated the test datasets to merge to single end Files: https://github.com/nf-core/test-datasets/commit/7a8d7d68c1ebb4f2ca0033c5d969f2801cb8c657

Let's see how the CI tests go now..

**Review by ewels — APPROVED** [review]
Tested locally and works great, thanks! 👍🏻 

Now also got the CI tests passing again, so I think we're good to merge 🚀
